test/install_test.go

```diff
@@ -261,3 +300,85 @@ func TestCheckProxy(t *testing.T) {
 		t.Fatal(err.Error())
 	}
 }
+
+func TestLogs(t *testing.T) {
+	controllerRegex := regexp.MustCompile("level=(panic|fatal|error|warn)")
+	proxyRegex := regexp.MustCompile(fmt.Sprintf("%s (ERR|WARN)", proxyContainer))
+
+	for deploy, spec := range linkerdDeployReplicas {
+		deploy := strings.TrimPrefix(deploy, "linkerd-")
+		containers := append(spec.containers, proxyContainer)
+
+		for _, container := range containers {
+			regex := controllerRegex
+			if container == proxyContainer {
+				regex = proxyRegex
+			}
+
+			outputStream, err := TestHelper.LinkerdRunStream(
+				"logs", "--no-color",
+				"--control-plane-component", deploy,
+				"--container", container,
+			)
+			if err != nil {
+				t.Fatalf("Error running command:\n%s", err)
+			}
+			defer outputStream.Stop()
+			outputLines, _ := outputStream.ReadUntil(100, 10*time.Second)
```

Comment:
For control plane containers that don't log at least 100 lines under normal testing conditions (e.g `tap` and `web`), this is going to add an additional 30 seconds to each test run (10 seconds each for standard install, tls, and single-namespace). And for other containers that log more than 100 lines, we won't be inspecting all of the logs emitted by the container. I think it would be better to fetch more log lines over a shorter window. How about:

```suggestion
			outputLines, _ := outputStream.ReadUntil(10000, time.Second)
```

Also, is skipping the error check here intentional since `outputLines` will be empty in the event of an error, and you're already checking that? Might be worth adding a comment.

Reply:
Yeah swallowing the error in case the timeout get hits (which now it consistently does). I've set the params to `(10000, 2*time.Second)`, with `(10000, 2*time.Second)` I was seeing it timeout before receiving any log lines.